### PR TITLE
Moving reserver/sourcer/defender management to room_external

### PR DESCRIPTION
**Pull request by ikiris**
I'm sure you're going to have me refactor things / add config options. I figured I'd save some of that till you can review since I have no idea style guidelines for this repo.

### CODE REVIEW

**Comment by democratic-collaboration** [comment]
[democratic collaboration](https://github.com/TooAngel/democratic-collaboration)
Approved reviews will speed up the merge, request changes will slow it down.

Please review the PR to help.

**Comment by democratic-collaboration** [comment]
Adding WIP label, the title is prefixed with [WIP]

**Comment by ikiris** [comment]
Realized some of the commit didn't get merged. Will finish it out tomorrow.

**Comment by TooAngel** [comment]
Had a quick look, looks fine so far

**Comment by democratic-collaboration** [comment]
Removing WIP label, the title is not prefixed with [WIP]

File: src/config.js [hunk 1]
```diff
@@ -130,7 +130,7 @@ global.config = {
     structurer: true,
     structurerInterval: 1500,
     structurerMinEnergy: 1300,
-    reserverDefender: true,
+    reserverDefender: false,
```
Comment (TooAngel) [inline src/config.js]: We should we disable it?

File: src/prototype_room_external.js [hunk 2]
```diff
@@ -277,33 +277,86 @@ Room.prototype.checkAndSpawnReserver = function() {
...
+
+  let idiotCreeps = _.filter(hostiles, {
     filter: this.findAttackCreeps
   });
   for (let idiotCreep of idiotCreeps) {
     brain.increaseIdiot(idiotCreep.owner.username);
   }
+};
+
+Room.prototype.checkSourcers = function() {
```
Comment (TooAngel) [inline src/prototype_room_external.js]: I think the logic for external rooms and `my` rooms is / should be the same.

https://github.com/TooAngel/screeps/blob/master/src/prototype_room_creepbuilder.js#L362

**Comment by democratic-collaboration** [comment]
A new review, yeah.

    Votes: 177/352
    Coefficient: 0.502840909091
    Merging in 1 days 9 hours
    Age 3 days 13 hours

**Comment by TooAngel** [comment]
Ah fixing the codeclimate issues, would be cool

**Comment by TooAngel** [comment]
Hey, resolving conflicts would be cool

**Comment by TooAngel** [comment]
Hey any chance to continue here or should we close the PR?

**Pull request closed without merging**